app/policies/collaborator_policy.rb

```diff
@@ -0,0 +1,11 @@
+class CollaboratorPolicy < ApplicationPolicy
+  def collaborator_search?
+    if record.collaboratorable_type == "User"
+      record.collaboratorable == user
+    elsif record.collaboratorable_type == "Jurisdiction"
+      user.jurisdictions.find(record.collaboratorable.id).present?
```

Comment:
I'm a bit concerned about database optimization on this line
is there any way to prevent excessive queries to jurisdictions and collaboratable?
if not then disregard

Reply:
Yeah realistically this should be fine. The collaboratorable will be called in the blueprint, rails should cache relationships for the same instance so this will just prevent an extra db call there. But you are right, I can just use the `collaboratorable_id` instead of `collaboratorable.id`. Will make the change. The jurisdiction query cannot be avoided